src/components/MapboxMap.tsx

```diff
@@ -0,0 +1,163 @@
+import { useRef, useEffect } from 'react';
+import mapboxgl, { Map, Marker, MapboxOptions, LngLatBounds, MarkerOptions, LngLat } from 'mapbox-gl';
+import { Result, useSearchState } from '@yext/search-headless-react';
+import { useDebouncedFunction } from '../hooks/useDebouncedFunction';
+import ReactDOM from 'react-dom';
+
+/**
+ * A functional component that can be used to render a custom marker on the map.
+ *
+ * @public
+ */
+export type PinComponent<T> = (props: { result?: Result<T>, index?: number }) => JSX.Element;
+
+/**
+ * A function use to derive a result's coordinate.
+ *
+ * @public
+ */
+export type CoordinateGetter<T> = (result: Result<T>) => Coordinate | undefined;
+
+/**
+ * Coordinate use to represent the result's location on a map.
+ *
+ * @public
+ */
+export interface Coordinate {
+  /** The latitude of the location. */
+  latitude: number,
+  /** The longitude of the location. */
+  longitude: number
+}
+
+/**
+ * Props for the {@link MapboxMap} component.
+ *
+ * @public
+ */
+export interface MapboxMapProps<T> {
+  /** Mapbox access token. */
+  mapboxAccessToken: string,
+  /** Interface for map customization derived from Mapbox GL's Map options. */
+  mapboxOptions?: Omit<MapboxOptions, 'container'>,
+  /** Custom Pin component to render for markers on the map. */
```

Comment:
can this doc mention that this component would replace the default pin from mapbox?